packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -22,45 +32,139 @@ const sidebarStyles = css({
+      <span>
+        There was a problem connecting{' '}
+        {info ? `to ${getConnectionTitle(info)}` : ''}
+      </span>
+      {info && (
+        <Link
+          className={connectionErrorToastActionMessageStyles}
+          hideExternalIcon={true}
+          onClick={onReview}
+        >
+          REVIEW
+        </Link>
+      )}
+    </span>
+  );
+}
 // Having props here is useful as a placeholder and we will fix it with the first props.
 // eslint-disable-next-line
-export function MultipleConnectionSidebar({}: MultipleConnectionSidebarProps) {
+export function MultipleConnectionSidebar({
+  appName,
+}: MultipleConnectionSidebarProps) {
+  const { openToast, closeToast } = useToast('multiple-connection-status');
+  const cancelCurrentConnectionRef = useRef<(id: string) => Promise<void>>();
+
+  const [isExpanded, setIsExpanded] = useState(true);
+  const [isConnectionFormOpen, setIsConnectionFormOpen] = useState(false);
+
+  const onConnected = useCallback(() => {
+    closeToast('connection-status');
+  }, [closeToast]);
+
+  const onConnectionAttemptStarted = useCallback(
+    (info: ConnectionInfo) => {
+      const cancelAndCloseToast = () => {
+        void cancelCurrentConnectionRef.current?.(info.id);
+        closeToast('connection-status');
+      };
+
+      openToast('connection-status', {
+        title: `Connecting to ${getConnectionTitle(info)}`,
+        dismissible: true,
+        variant: 'progress',
+        actionElement: (
+          <Link hideExternalIcon={true} onClick={cancelAndCloseToast}>
+            CANCEL
+          </Link>
+        ),
+      });
+    },
+    [openToast, closeToast, cancelCurrentConnectionRef]
+  );
+
+  const onConnectionFailed = useCallback(
+    (info: ConnectionInfo | null, error: Error) => {
+      const reviewAndCloseToast = () => {
+        closeToast('connection-status');
+        setIsConnectionFormOpen(true);
+      };
+
+      openToast('connection-status', {
+        title: `${error.message}`,
+        description: (
+          <ConnectionErrorToastBody
+            info={info}
+            onReview={reviewAndCloseToast}
+          />
+        ),
+        variant: 'warning',
+        timeout: 5000,
+      });
+    },
+    [openToast, closeToast, setIsConnectionFormOpen]
+  );
+
   const {
+    setActiveConnectionById,
     connect,
     favoriteConnections,
     recentConnections,
+    cancelConnectionAttempt,
     removeConnection,
     saveConnection,
+    duplicateConnection,
+    createNewConnection,
     state,
   } = useConnections({
-    onConnected: noop_tmp, // TODO: COMPASS-7710,
-    onConnectionAttemptStarted: noop_tmp,
-    onConnectionFailed: noop_tmp,
-    appName: '', // TODO: COMPASS-7710
-    getAutoConnectInfo: noop_tmp, // TODO: COMPASS-7710
+    onConnected: onConnected,
+    onConnectionAttemptStarted: onConnectionAttemptStarted,
+    onConnectionFailed(info, error) {
+      void onConnectionFailed(info, error);
+    },
+    appName: appName,
```

Comment:
It doesn't seem to be used by anything in the `useConnections`, why are we adding this?

Reply:
Ah, it was used before and probably kept it after the merge with Himanshu's PR and forgot to remove it. Thanks for pointing out!